apps/web/src/app/core/services/user.service.ts

```diff
@@ -0,0 +1,49 @@
+import { inject, Injectable } from '@angular/core';
+import { Auth } from '@angular/fire/auth';
+import { collection, collectionData, Firestore, orderBy, query, where, limit } from '@angular/fire/firestore';
+import type { Observable } from 'rxjs';
+import { map, of } from 'rxjs';
+
+import type { IActivity, IUser } from '../interfaces';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UserService {
+  readonly #authenticatedUser = inject(Auth).currentUser;
+  readonly #firestore = inject(Firestore);
+  readonly #path = 'users';
+
+  public getMe(): Observable<IUser | null> {
+    if (!this.#authenticatedUser) {
+      return of(null);
+    }
+    return this.getUser(this.#authenticatedUser.uid);
+  }
+
+  public getUser(uid: string): Observable<IUser | null> {
+    const queryResult = query(this.#collection(), where('uid', '==', uid)).withConverter({
+      fromFirestore: (snap) => snap.data() as IUser,
+      toFirestore: (data) => data,
+    });
+
+    return collectionData<IUser | null>(queryResult).pipe(map((users) => users[0] || null));
+  }
+
+  public getUserActivities(user: IUser, maxActivities = 100): Observable<IActivity[]> {
+    const queryResult = query(
+      this.#collection(`/users/${user.id}/activities`),
+      where('type', '!=', 'PR_UPDATED'),
```

Comment:
Another way to look at the filtering, for timebeing, is to filter on activities where the user gained Xp?

Reply:
That'll be a good improvement. I'll add it :-)